go/vt/vttablet/tabletmanager/vreplication/engine.go

```diff
@@ -98,15 +101,34 @@ type journalEvent struct {
 
 // NewEngine creates a new Engine.
 // A nil ts means that the Engine is disabled.
-func NewEngine(ts *topo.Server, cell string, mysqld mysqlctl.MysqlDaemon, dbClientFactory func() binlogplayer.DBClient, dbName string) *Engine {
+func NewEngine(config *tabletenv.TabletConfig, ts *topo.Server, cell string, mysqld mysqlctl.MysqlDaemon) *Engine {
+	dbClientFactory := func() binlogplayer.DBClient {
+		return binlogplayer.NewDBClient(config.DB.FilteredWithDB())
+	}
+	vre := &Engine{
+		controllers:     make(map[int]*controller),
+		ts:              ts,
+		cell:            cell,
+		mysqld:          mysqld,
+		dbClientFactory: dbClientFactory,
+		dbName:          config.DB.DBName,
+		journaler:       make(map[string]*journalEvent),
+		ec:              newExternalConnector(config.ExternalConnections),
+	}
```

Comment:
Is there a possibility of source being external mysql and the configs are not set properly. In that case the connector might not work as expected.

Reply:
This is possible. It's not very different from the normal case where a vttablet is not reachable. It will just keep retrying every 5s.

Reply:
I validated this behaves in that way.